### PR TITLE
Fixes #11921 - load dynamic features during runtime

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by dLobatog** [comment]
@ares Mind to write a test? I think it'll be hard to understand why this is here and might be removed in the future without it. Thanks - it works well.

**Comment by ares** [comment]
Test added, please re-review.

File: test/functional/smart_proxy_auth_test.rb [hunk 1]
```diff
@@ -35,6 +35,18 @@ def test_successful_smart_proxy_authentication_in_api_controller
...
+
+  def test_require_smart_proxy_or_login_accepts_callable_features
+    User.current = users(:admin)
+
+    @controller.stubs(:auth_smart_proxy).returns(false)
+    @controller.stubs(:require_login).returns(true)
+    @controller.stubs(:authorize).returns(true)
+
+    assert_raise ArgumentError, 'test' do
+      @controller.send(:require_smart_proxy_or_login, Proc.new { raise ArgumentError, 'test' } )
```
Comment (dLobatog) [inline test/functional/smart_proxy_auth_test.rb:47]: Interesting way of testing it's dynamic :smile:

**Comment by dLobatog** [comment]
Merged as 60195b1278bfaa236f3399b956b6dcbca21ef7c4, thanks @ares!

File: test/functional/smart_proxy_auth_test.rb (hunk 1 above)
Comment (ares) [inline test/functional/smart_proxy_auth_test.rb:47]: ;-)

**Pull request closed without merging**